api/client/webclient.go

```diff
@@ -0,0 +1,194 @@
+/*
+Copyright 2020-2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"net"
+	"net/http"
+	"net/url"
+
+	"github.com/gravitational/teleport/api/constants"
+	"github.com/gravitational/trace"
+)
+
+const (
+	invalidProxyAddressMessage = "'%v' is not a valid proxy address"
+)
+
+// initClient creates a new client to the HTTPS web proxy.
+func initClient(proxyAddr string) (*http.Client, error) {
+	// Validate proxyAddr.
+	host, port, err := net.SplitHostPort(proxyAddr)
+	if err != nil || host == "" || port == "" {
+		if err != nil {
+			return nil, trace.Wrap(err, invalidProxyAddressMessage, proxyAddr)
+		}
+		return nil, trace.BadParameter(invalidProxyAddressMessage, proxyAddr)
+	}
+	if _, err := url.Parse("https://" + net.JoinHostPort(host, port)); err != nil {
+		return nil, trace.BadParameter(invalidProxyAddressMessage, proxyAddr)
+	}
+
+	// Skip https cert verification, print a warning that this is insecure.
+	return NewInsecureWebClient(), nil
+}
+
+func NewInsecureWebClient() *http.Client {
+	return &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
+			},
+		},
+	}
+}
+
+// Ping serves two purposes. The first is to validate the HTTP endpoint of a
+// Teleport proxy. This leads to better user experience: users get connection
+// errors before being asked for passwords. The second is to return the form
+// of authentication that the server supports. This also leads to better user
+// experience: users only get prompted for the type of authentication the server supports.
+func Ping(ctx context.Context, proxyAddr string) (*PingResponse, error) {
```

Comment:
this was copied from `lib/client/weblogin.go`, so please remove their implementation and update all references to use this version